core/io/resource_format_binary.cpp

```diff
@@ -838,7 +838,8 @@ void ResourceInteractiveLoaderBinary::open(FileAccess *p_f) {
 	if (header[0] == 'R' && header[1] == 'S' && header[2] == 'C' && header[3] == 'C') {
 		//compressed
 		FileAccessCompressed *fac = memnew(FileAccessCompressed);
-		fac->open_after_magic(f);
+		error = fac->open_after_magic(f);
+		ERR_FAIL_COND_MSG(error != OK, "Failed to open binary resource file: " + local_path + ".");
```

Comment:
Aren't we going to leak `f` and `fac`?
I tried to solve this too a few weeks ago, attempting to properly free memory, and I was getting weird crashes :/

Reply:
See https://github.com/godotengine/godot/issues/32511#issuecomment-574062333

Reply:
Ah, yes, it will leak `fac`. I'll look further into the handling of `f`.